components/user/Tabs/SettingsContent.tsx

```diff
@@ -1,59 +1,260 @@
+import { UNSET } from 'utils/forms'
+import { useQueriedToast } from 'hooks/useToast'
+import { STATUS } from 'hooks/useLogin'
+
+import * as API from 'apiClient'
+import { TabType } from './index'
+
 type Props = {
-  authedUser: API.ApiUserMetadata
+  anyBlocksMined: boolean
+  user: API.ApiUser
+  authedUser: API.ApiUserMetadata | null
+  toast: ReturnType<typeof useQueriedToast>
+  reloadUser: () => Promise<boolean>
+  setUserStatus: (x: STATUS) => unknown
+  setRawMetadata: (x: API.ApiUserMetadata) => unknown
+  onTabChange: (tab: TabType) => unknown
+  setFetched: (x: boolean) => unknown
+  setUser: (x: API.ApiUser) => unknown
 }
 
-const getSocialChoice = (
-  authedUser: API.ApiUserMetadata
-): { choice: string; value: string } => {
-  if (authedUser.discord) {
-    return { choice: 'discord', value: authedUser.discord }
-  } else if (authedUser.telegram) {
-    return { choice: 'telegram', value: authedUser.telegram }
-  }
-  return { choice: '', value: '' }
+const EDITABLE_FIELDS = {
+  email: {
+    ...FIELDS.email,
+    validation: () => true,
+    touched: true,
+    controlled: true,
+  },
+  graffiti: { ...FIELDS.graffiti, controlled: true },
+  country: { ...FIELDS.country, useDefault: false, controlled: true },
+  discord: {
+    ...FIELDS.social,
+    required: false,
+    options: undefined,
+    id: 'discord',
+    label: 'Discord',
+    placeholder: 'Your Discord username',
+    isRadioed: false,
+    validation: () => true,
+    controlled: true,
+  },
+  telegram: {
+    ...FIELDS.social,
+    required: false,
+    options: undefined,
+    id: 'telegram',
+    label: 'Telegram',
+    placeholder: 'Your Telegram username',
+    isRadioed: false,
+    validation: () => true,
+    controlled: true,
+  },
 }
 
-export default function SettingsContent({ authedUser }: Props) {
-  const $email = useField(FIELDS.email)
-  const $graffiti = useField(FIELDS.graffiti)
-  const $social = useField(FIELDS.social)
-  const $country = useField(FIELDS.country)
+export default function SettingsContent({
+  anyBlocksMined,
+  user,
+  authedUser,
+  toast,
+  reloadUser,
+  onTabChange,
+  setFetched,
+  setUser,
+  setUserStatus,
+  setRawMetadata,
+}: Props) {
+  const router = useRouter()
+  const [$error, $setError] = useState<string>(UNSET)
+  const [$loading, $setLoading] = useState(false)
+  const {
+    email: _email = UNSET,
+    graffiti: _graffiti = UNSET,
+    discord: _discord = UNSET,
+    telegram: _telegram = UNSET,
+    country_code: _country_code = UNSET,
+  } = authedUser || {}
+
+  const $graffiti = useField({
+    ...EDITABLE_FIELDS.graffiti,
+    defaultValue: _graffiti,
+  })
+  const $email = useField({
+    ...EDITABLE_FIELDS.email,
+    defaultValue: _email,
+  })
+  const $discord = useField({
+    ...EDITABLE_FIELDS.telegram,
+    defaultValue: _telegram,
+    touched: !!_telegram,
+  })
+  const $telegram = useField({
+    ...EDITABLE_FIELDS.discord,
+    defaultValue: _discord,
+    touched: !!_discord,
+  })
```

Comment:
I just noticed the Discord and Telegram fields seem to be swapped, so updating one actually updates the other.

Reply:
Great catch. Addressed.